apps/arweave/src/ar_nonce_limiter.erl

```diff
@@ -449,6 +483,29 @@ handle_call({get_step_checkpoints, StepNumber, SessionKey}, _From, State) ->
 			{reply, maps:get(StepNumber, Map, not_found), State}
 	end;
 
+handle_call({get_step_checkpoints_seed_upper_bound, StepNumber, SessionKey}, _From, State) ->
+	case get_session(SessionKey, State) of
+		not_found ->
+			{reply, not_found, State};
+		#vdf_session{ step_checkpoints_map = Map, seed = Seed,
+				upper_bound = UpperBound, next_upper_bound = NextUpperBound } ->
+			case maps:get(StepNumber, Map, not_found) of
+				not_found ->
+					{reply, not_found, State};
+				L ->
+					{_NextSeed, IntervalNumber, _NextVDFDifficulty} = SessionKey,
+					IntervalStart = IntervalNumber * ?NONCE_LIMITER_RESET_FREQUENCY,
+					UpperBound2 =
+						case get_entropy_reset_point(IntervalStart, StepNumber) of
```

Comment:
Do we also need to choose Seed vs. NextSeed based on the entropy reset point here?

Reply:
No, we always use "Seed" to determine recall ranges.